astropy/io/fits/header.py

```diff
@@ -644,6 +644,10 @@ def tostring(self, sep='', endcard=True, padding=True):
             while s:
                 lines.append(s[:Card.length])
                 s = s[Card.length:]
+                # On the last CONTINUE card remove the final '&'
+                if s and len(s) <= Card.length:
+                    loc = s.rfind('&')
+                    s = s[:loc] + s[loc+1:] + ' '
```

Comment:
This is not bad as a first pass workaround to the problem.

But a better place to fix this would be in the code that actually string-formats cards, so that `str(card)` works properly as well.  You might want to look at some of the other unit tests like [this one](https://github.com/sornars/astropy/blob/continue-final-amp/astropy/io/fits/tests/test_header.py#L436) that test the string representation of an individual card, since that's all that matters *generally* when a Header is written (though it can't hurt to test it in the context of `str(header)` as well).

So anyways, a better place to go fixing this issue would be in the method [`Card._format_long_image`](https://github.com/sornars/astropy/blob/continue-final-amp/astropy/io/fits/card.py#L1254).  The `Card` class is pretty complicated (perhaps overly so) but you don't need to look much beyond that method, as it's where the logic of actually formatting a `CONTINUE` card is.  My preference would be to fix the code so that it doesn't write an `'&'` on the last card in the first place.  But this fix would be more complicated, admittedly, but I think necessary.  Your current fix of finding and stripping the last `'&'` on the last line is *almost* correct.  But unfortunately it does introduce a small regression where comments are involved.  Take for example:

```
In [3]: h = Header()

In [4]: h['TEST'] = ('long value' * 10, 'long comment &' * 10)

In [5]: h
Out[5]: 
TEST    = 'long valuelong valuelong valuelong valuelong valuelong valuelong &'  
CONTINUE  'valuelong valuelong valuelong value&'                                
CONTINUE  '&' / long comment &long comment &long comment &long comment &long    
CONTINUE  '&' / comment &long comment &long comment &long comment &long comment 
CONTINUE  '&' / &long comment &
```

The "comment" is the part of the last 3 lines that follow the `'/'`.  Your fix would incorrectly strip an ampersand out of the comment (it doesn't have any special meaning in comments, even for CONTINUE cards), but leave the `'&'` that continues the value.  So you'll have to fix the logic in `_format_long_image` to not write the `&` in the *value* portion of the last CONTINUE card.